CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_DefaultComplications.uc

```diff
@@ -0,0 +1,278 @@
+	Template.CanBeChosen = SupplyAndIntelChains;
+
+	return Template;
+}
+
+function SpawnRescueMission(XComGameState NewGameState, XComGameState_Complication ComplicationState)
+{
+	local XComGameState_Complication_RewardInterception ActualComplicationState;
+	local XComGameState_ActivityChain InterceptedChainState, SpawnedChainState;
+	local X2ActivityChainTemplate ChainTemplate;
+	local XComGameState_Activity ActivityState;
+	local X2StrategyElementTemplateManager TemplateManager;
+	local X2ActivityTemplate_Mission ActivityTemplate;
+	local XComGameState_ResourceContainer TotalResContainer;
+
+	TemplateManager = class'X2StrategyElementTemplateManager'.static.GetStrategyElementTemplateManager();
+
+	ActivityState = InterceptedChainState.GetLastActivity();
+	ActivityTemplate = X2ActivityTemplate_Mission(ActivityState.GetMyTemplate());
+
+	if (ActivityTemplate == none)
+	{
+		`RedScreen("Failed to find source activity!");
+		return;
+	}
+
+	if (ActivityTemplate.MissionRewards.Find('Reward_Intel') > -1)
+	{
+		ChainTemplate = X2ActivityChainTemplate(TemplateManager.FindStrategyElementTemplate('ActivityChain_IntelIntercept'));
+	}
+	else
+	{
+		ChainTemplate = X2ActivityChainTemplate(TemplateManager.FindStrategyElementTemplate('ActivityChain_SupplyIntercept'));
+	}
+
+	SpawnedChainState = ChainTemplate.CreateInstanceFromTemplate(NewGameState);
+	SpawnedChainState.FactionRef = InterceptedChainState.FactionRef;
+	SpawnedChainState.PrimaryRegionRef = InterceptedChainState.PrimaryRegionRef;
+
+	ActualComplicationState = XComGameState_Complication_RewardInterception(ComplicationState);
+	TotalResContainer = XComGameState_ResourceContainer(NewGameState.ModifyStateObject(class'XComGameState_ResourceContainer', ActualComplicationState.ResourceContainerRef.ObjectID));
+
+	TotalResContainer.CombineLoot();
+	SpawnedChainState.ChainObjectRefs.AddItem(TotalResContainer.GetReference());
+	SpawnedChainState.StartNextStage(NewGameState);
+}
+
+function bool SupplyAndIntelChains(XComGameState NewGameState, XComGameState_ActivityChain ChainState)
+{
+	local XComGameState_ActivityChain OtherChainState;
+	local XComGameState_Activity ActivityState;
+	local X2ActivityTemplate_Mission ActivityTemplate;
+
+	ActivityState = ChainState.GetLastActivity();
+	ActivityTemplate = X2ActivityTemplate_Mission(ActivityState.GetMyTemplate());
+
+	if (ActivityTemplate == none) return false;
+	
+	// if this chain doesn't have any other complications
+
+	if(ChainState.ComplicationRefs.Length != 0)
+	{
+		return false;
+	}
+	
+	// and if there is a supply or intel rewarding chain
+	if (IsLootcrateActivity(ActivityTemplate) || ActivityTemplate.MissionRewards.Find('Reward_Intel') > -1)
+	{
+		// and if no other chains already have this complication
+		foreach `XCOMHISTORY.IterateByClassType(class'XComGameState_ActivityChain', OtherChainState)
+		{
+			if (OtherChainState.bEnded == false && OtherChainState.HasComplication('Complication_RewardInterception'))
```

Comment:
Move these checks to flags on the complication template

* `bExclusiveOnChain`
* `bNoSimultaneous`